andela/travel_tool_back
# #162376440 Add checklist item validations

#### What does this PR do?
Ensures a user does not upload duplicate checklist items

#### Description of Task to be completed?
* Add checklist item validations to validate only unique items are uploaded

#### How should this be manually tested?
 1. Clone the repo
    `git clone https://github.com/andela/travel_tool_back.git`
 2. Switch to the project directory
   `cd travel_tool_back`
 3. Checkout the **ch-restrict-request-dates-overlap-162274568** branch
   `git checkout ch-validate-checklist-items-162376440`
 4. Rollback, run database migrations and seed database
   `yarn db:rollmigrate`
 5. Install dependencies and start the server
   `make start`
 6. Start the frontend of this project.
 7. Navigate to [travela](http://travela-local.andela.com:3000/) in your browser and log in.
 8. Make yourself a Manager in the database by adding another role in the `UserRoles` table with your user id and the roleId as 53019.
- Also, make yourself a travel Admin
 9. Make a request with yourself as your line manager. The request destination should be your center
 10. Approve that requests
 11. Navigate to `http://local.andela.com:3000/checklists` and create 2 travel checklist e.g `Passport` and `Visa`
12. Go to the requests you made and click on the elipsis icon. Choose `Travel checklist` and upload a visa and the passport.
13. Try to upload the same document twice. It should not allow you. You should see the results shown in the screenshot shown below/
#### Any background context you want to provide?
You will need the latest frontend of [this](https://github.com/andela/travel_tool_front/tree/eedbdc1c5fc9ad1b5bf27e24c051cc2842219ed3) project as today.

#### What are the relevant pivotal tracker stories?
 [#162376440](https://www.pivotaltracker.com/story/show/162376440)

#### Screenshots (if appropriate)
![ezgif com-video-to-gif 2](https://user-images.githubusercontent.com/22371596/49508289-5e410300-f893-11e8-98a8-4a971632106a.gif)
